docs/src/types/vectors.md

```diff
@@ -0,0 +1,177 @@
+# Vectors
+
+## Passing in vectors
+
+You can pass a Rust `std::vec::Vec` into your contract method transparently. The following code calls a Sway function which accepts and then returns a `Vec<SomeStruct<u32>>`.
+
+```rust,ignore
+{{#include ../../../packages/fuels/tests/harness.rs:passing_in_vec}}
+```
+
+You can use a vector just like you would use any other type -- i.e. a `[Vec<u32>; 2]` or a `SomeStruct<Vec<Bits256>>` etc.
+
+## Returning vectors
+
+There is a mandatory extra step to returning vectors -- you need to `log` every element before returning the vector.
+
+You must not mix any unrelated `log`s once you start logging vector elements. 
+
+These logs must be the last ones you make.
+
+
+### A simple case:
+
+```Rust
+contract;
+
+use std::logging::log;
+
+abi MyContract {
+    fn u32_vec(arg: Vec<u32>) -> Vec<u32>;
+}
+
+
+impl MyContract for Contract {
+    fn some_fn() -> Vec<u32> {
+        let mut a_vec = ~Vec::new();
+        a_vec.push(1);
+        a_vec.push(2);
+
+        log_vec(expected);
+        a_vec
+    }
+}
+
+fn log_vec<T>(vec: Vec<T>) {
+    let mut i = 0;
+    while i < vec.len() {
+        let el = vec.get(i);
+
+        match el {
+            Option::Some(val) => log(val),
+            _ => assert(false),
+        };
```

Comment:
Actually, now that I'm looking at these match statements, you should probably just use `unwrap()` here since you already know from the condition of the loop that `get()` will not return `None`.

Reply:
Funny thing. `Option<something>` doesn't have an `unwrap` method, at least that is what the compiler (v0.24.4) tells me. Perhaps I'll have more luck in `0.24.5`, will try it now.

Reply:
It should! https://github.com/FuelLabs/sway/blob/44f6bcb50cec49275a30afa23c5a0d5fde86c585/sway-lib-std/src/option.sw#L47
If that's not working, then there's a bug somewhere 🤔

Reply:
For this example:
```
contract;

abi MyContract {
    fn test_function();
}

impl MyContract for Contract {
    fn test_function() {
        let mut a_vec = ~Vec::new();
        a_vec.push(1);

        let el = a_vec.get(1).unwrap();
    }
}
```
This is the output on `0.24.4`:
```
❯ forc build
  Creating a new `Forc.lock` file. (Cause: lock file did not match manifest)
  Removing core
  Removing std git+https://github.com/fuellabs/sway?tag=v0.24.5#e695606d8884a18664f6231681333a784e623bc9
    Adding core
    Adding std git+https://github.com/fuellabs/sway?tag=v0.24.4#f7800e18979dd8e4c8d672b8e7aae1758c2edb88
   Created new lock file at /home/segfault_magnet/example/Forc.lock
  Compiled library "core".
  Compiled library "std".
error
  --> /home/segfault_magnet/example/src/main.sw:13:24
   |
11 |
12 |
13 | 	let el = a_vec.get(1).unwrap();
   |                       ^^^^^^ No method named "unwrap" found for type "Option<u64>".
14 |     }
15 | }
   |
____

  Aborting due to 1 error.
Error: Failed to compile example

✦ ~/example
```

But on 0.24.5 it works :partying_face: . So I'm going to be changing these to unwraps

Reply:
Ahhh... 

0.24.5 will work as well if you `use std::option::Option`.